### PR TITLE
add method to fetch data without decompressing

### PR DESCRIPTION
**Pull request by axos88**
We may want to do somthing with the `fn read()` as well

### CODE REVIEW

**Comment by tilpner** [comment]
Why do you want to fetch the data without decompressing it? Maybe there's a better solution than exposing the compressed raw data

File: lib/src/lib.rs [hunk 1]
```diff
@@ -51,31 +52,29 @@ impl Files {
...
+        }
+    }
+
+    pub fn get_raw(&self, path: &str) -> io::Result<(Compression, Cow<'static, [u8]>)> {
         if self.passthrough.load(Ordering::Relaxed) {
             let mut r = BufReader::new(File::open(path)?);
             let mut v = Vec::new();
             r.read_to_end(&mut v)?;
-            return Ok(Cow::Owned(v))
+            return Ok((Compression::None, Cow::Owned(v)));
```
Comment (tilpner) [inline lib/src/lib.rs]: This doesn't need the trailing semicolon

File: lib/src/lib.rs [hunk 2]
```diff
@@ -51,31 +52,29 @@ impl Files {
...
-                        Ok(Cow::Owned(v))
-                    }
-                    #[cfg(not(feature = "flate2"))]
-                    Compression::Gzip => panic!("Feature 'flate2' not enabled"),
-                }
-            }
-            None => Err(Error::new(ErrorKind::NotFound, "Key not found")),
-        }
+
+        self.files.get(key.borrow() as &str).map( |(c,d)| (*c, Cow::Owned((*d).into()))).ok_or(Error::new(ErrorKind::NotFound, "Key not found"))
```
Comment (tilpner) [inline lib/src/lib.rs]: This is way too long. How about:

```
        self.files.get(&*key)
            .map(|&(c, d)| (c, Cow::Owned(d.to_owned())))
            .ok_or(Error::new(ErrorKind::NotFound, "Key not found"))
```

File: lib/src/lib.rs [hunk 3]
```diff
@@ -51,31 +52,29 @@ impl Files {
     }
 
     pub fn get(&self, path: &str) -> io::Result<Cow<'static, [u8]>> {
+        match self.get_raw(path) {
+            Ok((Compression::None, data)) => Ok(data),
+            Ok((Compression::Gzip, compressed)) => {
+                let mut r = GzDecoder::new(Cursor::new(compressed));
```
Comment (tilpner) [inline lib/src/lib.rs:59]: There is no GzDecoder without flate2. I see you've added a #[cfg] in #8, but it really belongs here.

**Review by tilpner — CHANGES_REQUESTED** [review]
Thanks for the PR, should be fine to merge after a few small changes

**Comment by axos88** [comment]
rebased on top of the updated #7.
My intentions should be a bit more obvious now.

**Comment by axos88** [comment]
The reason for fetching data without decompressing it is... well, because I need the data without decompressing it :)

The goal is to serve the files over http as an asset, and without this change includedir decompresses it, only for the web framework to recompess it when passing it to the browser with (Content-Encoding: gzip).
With this change, I can provide the already compressed data stream directly to the framework to send it to the browser unchanged.

**Comment by axos88** [comment]
Thanks 👍

**Pull request merged**